File: W3/homeWork_W3/Java/PenPaper/PPMain.java
public class PPMain {
    public static void main(String[] args) {
        try {
            Pen pen = new Pen(400);
            Paper paper = new Paper(300);
            String str1 = "Hello World! I love C++ so much."; // 32
            String str2 = "\nBut he loves Java."; // 19
            
            pen.refill();
            pen.write(paper, str1);
            paper.show();
            
            System.out.println("inkAmount - " + pen.getInkAmount());
            System.out.println("inkCapacity - " + pen.getInkCapacity());
            System.out.println("symbols - " + paper.getSymbols());
            System.out.println("maxSymbols - " + paper.getMaxSymbols());
            
            pen.write(paper, str2);
            paper.show();
            
            System.out.println("inkAmount - " + pen.getInkAmount());
            System.out.println("inkCapacity - " + pen.getInkCapacity());
            System.out.println("symbols - " + paper.getSymbols());
            System.out.println("maxSymbols - " + paper.getMaxSymbols());
            
            pen.write(paper, str1);
            paper.show();
            
            System.out.println("inkAmount - " + pen.getInkAmount());
            System.out.println("inkCapacity - " + pen.getInkCapacity());
            System.out.println("symbols - " + paper.getSymbols());
            System.out.println("maxSymbols - " + paper.getMaxSymbols());
            
            // pen.write(paper, str);
            // paper.show();
            
            // cout << "inkAmount - " << pen.getInkAmount() << endl;
            // cout << "inkCapacity - " << pen.getInkCapacity() << endl;
            // cout << "symbols - " << paper.getSymbols() << endl;
            // cout << "maxSymbols - " << paper.getMaxSymbols() << endl;
        }
        catch (OutOfInkException exception) {
            System.out.println("Please, refill your pen!");
        }
        catch (OutOfSpaceException exception) {
            System.out.println("Not enough space! You used all free space on the paper!");
        }
    }
}

File: PHP/PenPaper/PPMain.php
<?php
    require_once 'Pen.php';
    
    try {
        $pen = new Pen(25);
        $paper = new Paper(12);
        
        $pen->write($paper, "Hello ");
        $pen->write($paper, "world!");
        $paper->show();
        
        try {
            $pen->write($paper, " I love PHP!");    
        
        } catch ( OutOfInk $ex ) {
            echo $ex->getMessage() . PHP_EOL;
        } catch ( OutOfSpace $ex ) {
            echo $ex->getMessage() . PHP_EOL;
        } 
        $paper->show();
        
    } catch ( InvalidParam $ex ) {
        echo $ex->getMessage() . PHP_EOL;
    } catch ( InvalidAttribute $ex ) {
        echo $ex->getMessage() . PHP_EOL;
    } catch ( OutOfInk $ex ) {
        echo $ex->getMessage() . PHP_EOL;
    } catch ( OutOfSpace $ex ) {
        echo $ex->getMessage() . PHP_EOL;
    }
?>
